lib/orbit_web/channels/user_socket.ex

```diff
@@ -0,0 +1,90 @@
+defmodule OrbitWeb.UserSocket do
+  @moduledoc """
+  Will always allow connection, even if user is unauthenticated or out of date,
+  So that we can tell the client what the issue is and they can handle it (by refreshing).
+
+  If we refuse unauthenticated connections, then the client can't tell the difference between
+  an auth problem where they should log in again, and a network problem where they should reconnect.
+  Phoenix has a way to reply with a connection error, and it reaches the browser, but not in a
+  way that's visible to the frontend application code.
+
+  After connection, clients should connect to the `metadata` channel to learn whether they were
+  authenticated or not.
+
+  All other channels should remember to check the release version and
+  authentication/authorization before allowing a client to join the channel.
+  """
+
+  use Phoenix.Socket
+  require Logger
+
+  channel "metadata", OrbitWeb.MetadataChannel
+
+  @doc """
+  fields that are added to socket.assigns
+  """
+  @type metadata :: %{
+          server_release: String.t(),
+          client_release: String.t() | nil,
```

Comment:
`client_release` could be nil? is that for running locally?

Reply:
Locally, the client release is always `local` (and reloading is handled by phoenix instead). https://github.com/mbta/orbit/blob/5c972eafb88ce811f585bd58c953731a2702fba7/config/dev.exs#L8

This is if you join the socket without passing in the client release. That doesn't have to be allowed, it's not part of any expected failure modes, but I was trying to make it so there were no application-defined rejections, so that any socket rejections were definitely because of network problems.